From: =?UTF-8?q?Picca=20Fr=C3=A9d=C3=A9ric-Emmanuel?= <picca@debian.org>
Date: Sun, 20 Oct 2013 08:48:49 +0200
Subject: feature forwarded add the build_doc target

---
 setup.py | 31 ++++++++++++++++++++++++++++++-
 1 file changed, 30 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 556ac85..7376d5f 100644
--- a/setup.py
+++ b/setup.py
@@ -64,6 +64,8 @@ class build_ext_subclass( build_ext ):
                 e.extra_link_args = lopt[ c ]
         build_ext.build_extensions(self)
 
+cmdclass = {'build_ext': build_ext_subclass}
+
 with open('README.txt') as f:
     long_description = f.read()
 
@@ -76,6 +78,33 @@ extmodul = Extension('xrayutilities.cxrayutilities',
                                 os.path.join('xrayutilities','src','gridder3d.c')],
                      define_macros = user_macros)
 
+try:
+    import sphinx
+    import sys
+
+    from sphinx.setup_command import BuildDoc
+
+    class build_doc(BuildDoc):
+        def run(self):
+            # make sure the python path is pointing to the newly built
+            # code so that the documentation is built on this and not a
+            # previously installed version
+            build = self.get_finalized_command('build')
+            sys.path.insert(0, os.path.abspath(build.build_lib))
+            try:
+                sphinx.setup_command.BuildDoc.run(self)
+            except UnicodeDecodeError:
+                print("ERROR: unable to build documentation"
+                      " because Sphinx do not handle"
+                      " source path with non-ASCII characters. Please"
+                      " try to move the source package to another"
+                      " location (path with *only* ASCII characters)")
+            sys.path.pop(0)
+
+    cmdclass['build_doc'] = build_doc
+except ImportError:
+    pass
+
 setup(name="xrayutilities",
       version="1.0.2",
       author="Eugen Wintersberger, Dominik Kriegner",
@@ -96,7 +125,7 @@ setup(name="xrayutilities",
       requires=['numpy','scipy','matplotlib','tables'],
       include_dirs = [numpy.get_include()],
       ext_modules = [extmodul],
-      cmdclass = {'build_ext': build_ext_subclass },
+      cmdclass = cmdclass,
       url="http://xrayutilities.sourceforge.net",
       license="GPLv2",
       script_args = args
