From: Eugen Wintersberger <eugen.wintersberger@gmail.com>
Date: Sat, 25 Apr 2015 13:18:32 +0200
Subject: Removed the logo_only option from the documentation build

---
 doc/source/conf.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/source/conf.py b/doc/source/conf.py
index 8271825..3c7f510 100644
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -109,7 +109,7 @@ html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-html_theme_options = {'logo_only': True}
+#html_theme_options = {'logo_only': True}
 
 # Add any paths that contain custom themes here, relative to this directory.
 # html_theme_path = []
