From: Eugen Wintersberger <eugen.wintersberger@gmail.com>
Date: Sat, 25 Apr 2015 22:00:14 +0200
Subject: Removed mathjax extension from sphinx build

---
 doc/source/conf.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/doc/source/conf.py b/doc/source/conf.py
index 6c68735..a09eca5 100644
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -35,7 +35,6 @@ extensions = [
     'sphinx.ext.todo',
     'sphinx.ext.coverage',
     'sphinx.ext.pngmath',
-    'sphinx.ext.mathjax',
     'sphinx.ext.viewcode',
     'rst2pdf.pdfbuilder']
 
